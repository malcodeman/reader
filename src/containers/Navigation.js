import React, { Component } from "react";
import styled from "styled-components";
import { connect } from "react-redux";

import { changeActiveLink } from "../actions";
import { color } from "../styles/constants";

const Nav = styled.nav`
  grid-area: n;
  background-color: #fff;
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  @media (min-width: 576px) {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background-color: #eee;
    flex-wrap: no-wrap;
  }
`;

const Link = styled.a`
  color: ${color.secondary};
  cursor: pointer;
  padding: 1rem;
`;

class Navigation extends Component {
  render() {
    return (
      <Nav>
        <Link onClick={() => this.props.changeActiveLink("reddit")}>Reddit</Link>
        <Link onClick={() => this.props.changeActiveLink("hacker_news")}>HN</Link>
      </Nav>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return{
    changeActiveLink: (link) => dispatch(changeActiveLink(link))
  }
};

export default connect(null, mapDispatchToProps)(Navigation);
