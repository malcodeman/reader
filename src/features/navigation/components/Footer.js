import React from "react";
import styled from "styled-components";

const StyledFooter = styled.footer`
  color: ${props => props.theme.secondary};
  margin-top: auto;
  padding: 1rem;
  font-size: 0.8rem;
`;

const Link = styled.a`
  color: #337ab7;
`;

const RepoLink = styled.span`
  display: none;
  @media (min-width: 576px) {
    display: inline;
  }
`;

const Footer = () => {
  return (
    <StyledFooter>
      Made by{" "}
      <Link href="https://github.com/malcodeman" target="_blank" rel="noopener">
        malcodeman{" "}
        <span aria-label="Woman Astronaut" role="img">
          👩‍🚀
        </span>
      </Link>
      <RepoLink>
        {" "}
        -{" "}
        <Link
          href="https://github.com/malcodeman/reader"
          target="_blank"
          rel="noopener"
        >
          repo
        </Link>
      </RepoLink>
    </StyledFooter>
  );
};

export default Footer;
