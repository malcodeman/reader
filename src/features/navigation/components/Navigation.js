import React from "react";
import styled from "styled-components";
import { NavLink } from "react-router-dom";

const StyledLink = styled(NavLink)`
  color: ${props => props.theme.secondary};
  cursor: pointer;
  padding: 1rem;
  text-decoration: none;
  &.active {
    color: ${props => props.theme.primary};
  }
`;

export const Navigation = () => {
  return (
    <>
      <StyledLink to="/" exact>
        All in one
      </StyledLink>
      <StyledLink to="/reddit">Reddit</StyledLink>
      <StyledLink to="/hn">HN</StyledLink>
    </>
  );
};

export default Navigation;
