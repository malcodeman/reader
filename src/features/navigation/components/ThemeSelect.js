import React from "react";
import { useDispatch, useSelector } from "react-redux";
import styled from "styled-components";

import { toggleDarkMode } from "../../settings/actions/actions_settings";

const Wrapper = styled.div`
  color: ${props => props.theme.secondary};
  padding: 1rem;
  font-size: 0.8rem;
`;

const EmojiWrapper = styled.span`
  cursor: pointer;
`;

const ThemeSelect = () => {
  const dispatch = useDispatch();
  const darkMode = useSelector(state => state.settings.darkMode);

  function toggle() {
    if (darkMode) {
      dispatch(toggleDarkMode(false));
    } else {
      dispatch(toggleDarkMode(true));
    }
  }

  return (
    <Wrapper>
      {darkMode ? (
        <EmojiWrapper>
          <span onClick={toggle} role="img" aria-label="light">
            ☀️
          </span>
        </EmojiWrapper>
      ) : (
        <EmojiWrapper>
          <span onClick={toggle} role="img" aria-label="dark">
            🌑
          </span>
        </EmojiWrapper>
      )}
    </Wrapper>
  );
};

export default ThemeSelect;
