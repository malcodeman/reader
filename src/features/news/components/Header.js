import React from "react";
import styled from "styled-components";

import { color } from "../../../state/styles/constants";

const Title = styled.h1`
  padding: 1rem;
  margin: 0;
  color: ${color.primary};
  display: none;
  text-transform: capitalize;
  @media (min-width: 576px) {
    display: block;
  }
`;

const Header = props => (
  <header>
    <Title>{props.title}</Title>
  </header>
);

export default Header;
