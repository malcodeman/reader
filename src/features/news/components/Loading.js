import React from "react";
import styled from "styled-components";

const StyledLoading = styled.span`
  padding: 1rem;
  color: ${props => props.theme.secondary};
  display: block;
`;

const Loading = props => {
  return <StyledLoading>{props.children}</StyledLoading>;
};

export default Loading;
