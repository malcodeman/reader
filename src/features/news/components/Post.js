import React from "react";
import styled from "styled-components";

import { color } from "../../../state/styles/constants";

const Link = styled.a`
  text-decoration: none;
  display: block;
  cursor: pointer;
`;

const Title = styled.p`
  margin: 0;
  color: ${color.primary};
`;

const common = `
  font-size: 0.8rem;
  line-height: 1.4;
  color: ${color.secondary};
  display: block;
  @media (min-width: 576px) {
    display: inline;
  }
`;

const Upvotes = styled.span`
  ${common} margin-right: 0.4rem;
`;

const Author = styled.span`
  ${common} margin-right: 0.4rem;
`;

const Comments = styled.span`
  ${common};
`;

const Post = props => {
  return (
    <Link href={props.url}>
      <Title>{props.title}</Title>
      <Upvotes>{props.upvotes} ups</Upvotes>
      <Author>by {props.author}</Author>
      <Comments>{props.comments} comments</Comments>
    </Link>
  );
};

export default Post;
