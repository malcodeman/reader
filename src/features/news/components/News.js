import React, { Component } from "react";
import styled from "styled-components";
import { connect } from "react-redux";

import Post from "./Post";
import Header from "./Header";
import { requestPosts } from "../actions/actions_news";

const NewsArea = styled.main`
  grid-area: m;
  background-color: transparent;
  @media (min-width: 576px) {
    padding: 2rem;
    overflow-y: scroll;
  }
`;

const Loading = styled.span`
  padding: 1rem;
  color: ${props => props.theme.secondary};
  display: block;
`;

const List = styled.ul`
  padding: 0;
  margin: 0;
  list-style: none;
`;

const ListItem = styled.li`
  padding: 1rem;
  :hover {
    background-color: #eee;
  }
`;

class News extends Component {
  componentDidMount = () => {
    this.props.requestPosts();
  };
  render() {
    return (
      <NewsArea>
        <Header title={this.props.source} />
        {this.props.loading ? (
          <Loading>Loading...</Loading>
        ) : (
          <List>
            {this.props.posts.map(post => (
              <ListItem key={post.id}>
                <Post
                  url={post.url}
                  title={post.title}
                  upvotes={post.upvotes}
                  author={post.author}
                  comments={post.comments}
                />
              </ListItem>
            ))}
          </List>
        )}
      </NewsArea>
    );
  }
}

const mapStateToProps = state => {
  return {
    source: state.news.source,
    loading: state.news.loading,
    posts: state.news.posts
  };
};

const mapDispatchToProps = dispatch => {
  return {
    requestPosts: () => dispatch(requestPosts())
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(News);
