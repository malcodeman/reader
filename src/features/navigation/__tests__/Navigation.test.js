import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { Navigation } from "../components/Navigation";

test("Navigation snapshot", () => {
  const wrapper = shallow(<Navigation />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
