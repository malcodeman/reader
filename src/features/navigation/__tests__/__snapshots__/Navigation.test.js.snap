// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation snapshot 1`] = `
<Fragment>
  <Styled(NavLink)
    exact={true}
    to="/"
  >
    All in one
  </Styled(NavLink)>
  <Styled(NavLink)
    to="/reddit"
  >
    Reddit
  </Styled(NavLink)>
  <Styled(NavLink)
    to="/hn"
  >
    HN
  </Styled(NavLink)>
</Fragment>
`;
